roadmap/implementers-guide/src/runtime/inclusion.md

```diff
@@ -68,13 +68,18 @@ All failed checks should lead to an unrecoverable error making the block invalid
   1. Transform each [`CommittedCandidateReceipt`](../types/candidate.md#committed-candidate-receipt) into the corresponding [`CandidateReceipt`](../types/candidate.md#candidate-receipt), setting the commitments aside.
   1. check the backing of the candidate using the signatures and the bitfields, comparing against the validators assigned to the groups, fetched with the `group_validators` lookup.
   1. check that the upward messages, when combined with the existing queue size, are not exceeding `config.max_upward_queue_count` and `config.watermark_upward_queue_size` parameters.
+  1. call `Router::ensure_processed_downward_messages(para, processed_downward_messages)` to check rules of processing the downward message queue.
+  1. check that the horizontal messages are sorted by ascending recipient ParaId and there is no two horizontal messages have the same recipient.
+  1. using `Router::ensure_horizontal_messages_fit(sender, horizontal_messages)` ensure that the sender para doesn't overfill any downward queue.
```

Comment:
This was renamed to `ensure_downward_messages_fit` in the `Router` module, right?

Reply:
nvm, now I see that it wasn't. the preamble of the `Router` message saying that there are two routines is kind of confusing.

Reply:
I rewrote and renamed routines `queue_downward_messages` and `ensure_downward_messages_fit`. Hope now it is a bit more clear.